### PR TITLE
feat: raplce okx contract

### PR DESCRIPTION
**Pull request by timelessover**
replace OKXMarket adapter for EVM chains

### CODE REVIEW

**Review by NedyalkoYotov — APPROVED** [review]

**Comment by liutaurasjanuskas** [comment]
Hello, I see that you've updated original provider. Please create a new one instead of updating the old one.

**Comment by timelessover** [comment]
> Hello, I see that you've updated original provider. Please create a new one instead of updating the old one.

How to do this?

**Comment by timelessover** [comment]
> Hello, I see that you've updated original provider. Please create a new one instead of updating the old one.

updated done, pls merge code

**Comment by liutaurasjanuskas** [comment]
Every new provider must have new folder/abi/index.ts file. Check documentation here https://dappradar.github.io/nft-sales-adapters/#/

**Comment by timelessover** [comment]
> Every new provider must have new folder/abi/index.ts file. Check documentation here https://dappradar.github.io/nft-sales-adapters/#/

ok. I created done

File: src/adapters/okx-avalanche-2/index.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    range: number;
+    chunkSize: number;
+    sdk: any;
+
+    constructor() {
+        this.name = "okx-avalanche-1";
+        this.protocol = "avalanche";
+        this.block = 36224739;
+        // this.deprecatedAtBlock = 26572425;
+        this.contract = "0xa7FD99748cE527eAdC0bDAc60cba8a4eF4090f7c";
```
Comment (liutaurasjanuskas) [inline src/adapters/okx-avalanche-2/index.ts:30]: Contracts/addresses must be lowercased for all EVM chains.

File: src/adapters/okx-bsc-2/index.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    range: number;
+    chunkSize: number;
+    sdk: any;
+
+    constructor() {
+        this.name = "okx-bsc-1";
+        this.protocol = "binance-smart-chain";
+        this.block = 32526441;
+        // this.deprecatedAtBlock = 16625257;
+        this.contract = "0xa7FD99748cE527eAdC0bDAc60cba8a4eF4090f7c";
```
Comment (liutaurasjanuskas) [inline src/adapters/okx-bsc-2/index.ts:30]: Contracts/addresses must be lowercased for all EVM chains.

File: src/adapters/okx-ethereum-2/index.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    range: number;
+    chunkSize: number;
+    sdk: any;
+
+    constructor() {
+        this.name = "okx-ethereum-1";
+        this.protocol = "ethereum";
+        this.block = 18311795;
+        // this.deprecatedAtBlock = 16625257;
+        this.contract = "0xa7FD99748cE527eAdC0bDAc60cba8a4eF4090f7c";
```
Comment (liutaurasjanuskas) [inline src/adapters/okx-ethereum-2/index.ts:30]: Contracts/addresses must be lowercased for all EVM chains.

File: src/adapters/okx-polygon-2/index.ts [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+];
+
+export const handleExtraData = (data: any) => {
+    const extraData = data.slice(2);
+    const singleOrderLength = extraDataConfig.reduce((pre, cur) => pre + cur.unit, 0) * gap;
+    let groupCursor = 0;
+    // split multiple order
+    const orderGroup = new Array(extraData.length / singleOrderLength).fill("").map((_, i) => {
+        let res = "";
+        res = extraData.substring(groupCursor, singleOrderLength * (i + 1));
```
Comment (liutaurasjanuskas) [inline src/adapters/okx-polygon-2/index.ts:68]: You can use single-line here (e.g. `const res = extraData...`)

File: src/adapters/okx-polygon-2/index.ts [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    range: number;
+    chunkSize: number;
+    sdk: any;
+
+    constructor() {
+        this.name = "okx-polygon-1";
+        this.protocol = "polygon";
+        this.block = 48612325;
+        // this.deprecatedAtBlock = 39539879;
+        this.contract = "0xa7FD99748cE527eAdC0bDAc60cba8a4eF4090f7c";
```
Comment (liutaurasjanuskas) [inline src/adapters/okx-polygon-2/index.ts:107]: Contracts/addresses must be lowercased for all EVM chains.

**Comment by timelessover** [comment]
All issues Done

File: src/adapters/okx-avalanche-2/index.ts [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    deprecatedAtBlock: number;
+    contract: string;
+    events: string[];
+    pathToAbi: string;
+    range: number;
+    chunkSize: number;
+    sdk: any;
+
+    constructor() {
+        this.name = "okx-avalanche-1";
```
Comment (liutaurasjanuskas) [inline src/adapters/okx-avalanche-2/index.ts:26]: Provider name must match with the folder name (e.g. this should be changed to `okx-avalanche-2`). Please do it with other providers that you have added.

File: src/adapters/okx-bsc-2/index.ts [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+    chunkSize: number;
+    sdk: any;
+
+    constructor() {
+        this.name = "okx-bsc-1";
+        this.protocol = "binance-smart-chain";
+        this.block = 32526441;
+        // this.deprecatedAtBlock = 16625257;
+        this.contract = "0xa7fd99748ce527eadc0bdac60cba8a4ef4090f7c";
+        this.events = ["MatchOrderResults"];
```
Comment (liutaurasjanuskas) [inline src/adapters/okx-bsc-2/index.ts:31]: There's no function `MatchOrderResults` in the ABI

**Comment by timelessover** [comment]
pls check again

**Review by liutaurasjanuskas — APPROVED** [review]

**Pull request merged**